sdk/src/azure/core/az_http_policy.c

```diff
@@ -67,8 +69,7 @@ AZ_NODISCARD az_result az_http_pipeline_policy_telemetry(
     remainder = az_span_copy_u8(remainder, '/');
     remainder = az_span_copy(remainder, AZ_SPAN_FROM_STR(AZ_SDK_VERSION_STRING));
 
-    telemetry_id = az_span_slice(
-        telemetry_id, 0, (int32_t)(az_span_ptr(remainder) - az_span_ptr(telemetry_id)));
+    telemetry_id = az_span_slice(telemetry_id, 0, _az_span_diff(remainder, telemetry_id));
```

Comment:
If I understand this correctly, this is just trimming away the extra slack space, correct, so the span length is just as large as the content we put in the buffer?

Reply:
yes, exactly